Frontend/ui-customer/src/Flow.purs

```diff
@@ -831,6 +833,96 @@ homeScreenFlow = do
               void $ pure $ toast (getString STR.SOMETHING_WENT_WRONG_PLEASE_TRY_AGAIN)
               modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props { currentStage = SearchLocationModel } })
             currentFlowStatus
+
+    EDIT_DESTINATION_SOFT state -> do
+      resp <- lift $ lift $ HelpersAPI.callApi $ Remote.makeEditLocationRequest state.data.driverInfoCardState.rideId state.props.destinationLat state.props.destinationLong state.data.destinationAddress
+      case resp of
+        Right (EditLocationRes editDestinationSoftResp) -> do
+          if (editDestinationSoftResp.bookingUpdateRequestId == Nothing) then do
+            void $ pure $ toast (getString STR.SOMETHING_WENT_WRONG_PLEASE_TRY_AGAIN)
+            modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{data{destination = state.data.driverInfoCardState.destination, destinationAddress = state.data.driverInfoCardState.destinationAddress}, props{destinationLat = state.data.driverInfoCardState.destinationLat, destinationLong = state.data.driverInfoCardState.destinationLng}})
```

Comment:
Can we move this `modifyState` in a function as its being used in multiple places